spec/dynamoid/criteria/chain_spec.rb

```diff
@@ -81,15 +81,26 @@ def request_params
         expect(model.where(request_params.merge({ name: 'Josh' })).batch(3).count).to eq(10)
       end
 
-      it 'supports combined limits' do
+      it 'supports combined limits with batch size 1' do
         # Scanning through 13 means it'll see 10 Josh objects and then
-        # 3 Pascal objects to which it'll return 2 since the record_limit will
-        # restrict it
+        # 3 Pascal objects but it'll hit record_limit first with 2 objects
+        # so we'd only see 12 requests due to batching.
+        expect(Dynamoid.adapter.client).to receive(request_type).exactly(12).times.and_call_original
```

Comment:
Do we test batch loading for `Scan`? Adapter is tested well but chain is not. We need at least smoky tests for limits and batch loading